examples/demo/index.rst

```diff
@@ -0,0 +1 @@
+.. image:: traits_ui_demo.jpg
```

Comment:
I am wondering if `tutorials` should look less polished than demo in the mean time. See #862 for merging the `demo` and `tutorials`.

Reply:
I agree that `demo` and `tutorials` should be merged, but I don't think we should avoid improving `tutorials` if it's easy to do so. The issue might be open for a while and somebody might be using tutorials app in the meantime.

Reply:
I was close to putting those files in for the reason you said, but then it would make `tutorials` look more polished, which I doubt it should be.

In the mean time, running the demo app with the tutorials gives a blank screen. The only difference adding an `index.rst` and an image would make is to make the `tutorials` look more like a polished product - one could say that is "improvement" if the tutorials is intended to be polished, or a misleading statement leading to disappointments if it is not intended to be.

I am not massively against adding them, this would be orthogonal to this PR though. The `demo` folder already has an image so it is within the scope of this PR.

Reply:
I don't think I understand the reasons for making `tutorials` look unpolished. Do you want to discourage people from using it so that it can be deleted later? In that case I would argue that `tutorials` should be the more polished one and have `demo` merged into it because then both Traits and TraitsUI would have matching example setups.

And I don't think adding the image is really orthogonal to this PR because `tutorials` technically did have the hardcoded image, it just failed to load it.

Reply:
Having two very similar code base is confusing to the developers. I can't say for sure which one should be made more polished. In fact, I was probably wrong in using the word "polish".  Polish-or-not is actually not the question here. The question is which folder was intended as the source of data for the demo application. At the moment, `demo` is intended for the demo application because it has all the detailed descriptions in the docstring.

Previously launching the application with `tutorials` produces HTML pointing to an image that can't be loaded, which is another indication that the `tutorials` folder was not the intended source of example files. This PR fixes that broken link by showing a blank screen instead. 

I am happy to defer that discussion to consider making the tutorials folder the main demo entry point instead. And please feel free to open a separate PR to add the image.

Reply:
Okay, I guess it makes sense that since the image wasn't in `tutorials` folder it isn't added to the application.